plugins/inputs/unbound/unbound.go

```diff
@@ -106,6 +94,12 @@ func (s *Unbound) Gather(acc telegraf.Accumulator) error {
 					fieldsThreads[threadID][field] = fieldValue
 				}
 			}
+		} else if suffix, found := strings.CutPrefix(stat, "histogram."); found {
+			if s.Histogram {
+				suffix, _, _ := strings.Cut(suffix, ".to.")
+				suffix = strings.TrimLeft(suffix, "0")
+				fields["histogram_"+suffix] = fieldValue
```

Comment:
I think we should keep the leading zero for numbers like `0.123` as they are now reported as `.123`...
```suggestion
				suffix = strings.TrimLeft(suffix, "0")
				if strings.HasPrefix(suffix, ".") {
					suffix = "0"+suffix
				}
				fields["histogram_"+suffix] = fieldValue
```

Reply:
I agree the leading zeros should be kept, I only droppped them because  [you suggested it](https://github.com/influxdata/telegraf/pull/16452#discussion_r1953031300). I see you have already committed this, thanks.